var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -104,13 +98,23 @@ def cmake_args(self):
         else:
             options.append('-DGMX_GPU:BOOL=OFF')
 
-        simd_value = self.spec.variants['simd'].value
-        if simd_value == 'auto':
-            pass
-        elif simd_value == 'none':
+        simd_features = {
+            'sse2': 'SSE2',
+            'sse4_1': 'SSE4.1',
+            'avx128fma': 'AVX_128_FMA',
+            'avx': 'AVX_256',
+            'axv128': 'AVX2_128',
+            'avx2': 'AVX2_256',
+            'avx512': 'AVX_512',
+            'avx512er': 'AVX_512_KNL',
+            'neon': 'ARM_NEON_ASIMD'
+        }
+        for feature in simd_features:
+            if feature in self.spec.target:
+                options.append('-DGMX_SIMD={0}'.format(simd_features[feature]))
```

Comment:
I think there's a bug in the logic here. Since most architectures match more than one SIMD instruction set we'll be appending multiple times `-DGMX_SIMD=<something>`. Also the order of iteration in a dictionary cannot be relied upon, at least across the Python versions Spack supports.